common/lib/xmodule/xmodule/modulestore/courseware_index.py

```diff
@@ -0,0 +1,133 @@
+""" Code to allow module store to interface with courseware index """
+from __future__ import absolute_import
+
+import logging
+
+from django.utils.translation import ugettext as _
+from opaque_keys.edx.locator import CourseLocator
+from search.search_engine_base import SearchEngine
+
+from . import ModuleStoreEnum
+from .exceptions import ItemNotFoundError
+
+# Use default index and document names for now
+INDEX_NAME = "courseware_index"
+DOCUMENT_TYPE = "courseware_content"
+
+log = logging.getLogger('edx.modulestore')
+
+
+class SearchIndexingError(Exception):
+    """ Indicates some error(s) occured during indexing """
+
+    def __init__(self, message, error_list):
+        super(SearchIndexingError, self).__init__(message)
+        self.error_list = error_list
+
+
+class ModuleStoreCoursewareIndexMixin(object):
+    """
+    Mixin class to enable indexing for courseware search from different modulestores
+    """
+
+    def do_index(self, location, delete=False):
+        """
+        Add to courseware search index from given location and its children
+        """
+        error_list = []
+        # TODO - inline for now, need to move this out to a celery task
```

Comment:
I apologise, you are quite correct - we created that as a result from other feedback - it should only be raised in case of a full-course index (manually initiated), and not in the automatic processing. I shall address.